File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] st = path.split("/");
        Stack<String>sta = new Stack<>();
        for (String s : st) {
            if (s.equals("") || s.equals(".")) {
                continue;
            } else if (s.equals("..")) {
                if (!sta.isEmpty()) {
                    sta.pop();
                }
            } else {
                sta.push(s);
            }
        }
        StringBuilder sb = new StringBuilder("/");
            for (String s : sta) {
                sb.append(s);
                sb.append("/");
            }

            if (sb.length() > 1) {
                sb.deleteCharAt(sb.length() - 1);
            }

            return sb.toString();
    }
}

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path=path.split("/")
        path=[i for i in path if i!=""]
        print(path)
        l=[]
        for i in path:
            if i==".":
                continue
            if i!='..':
                l.append(i)
            else:
                if l:
                    l.pop()
                else:
                    continue
        
        a="/"+"/".join(map(str,l))
        
        print(a)
        return a